client/state/data-layer/wpcom/signup/segments/test/index.js

```diff
@@ -0,0 +1,37 @@
+/** @format */
+
+/**
+ * Internal dependencies
+ */
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { requestSegments, storeSegments, showSegmentsRequestError } from '../';
+import { setSegments } from 'state/signup/segments/actions';
+import { NOTICE_CREATE } from 'state/action-types';
+
+describe( 'data-layer/wpcom/signup/segments', () => {
+	test( 'requestSegments()', () => {
+		const mockAction = {};
+		expect( requestSegments( mockAction ) ).toEqual(
+			http(
+				{
+					apiNamespace: 'wpcom/v2',
+					method: 'GET',
+					path: '/segments',
+				},
+				mockAction
+			)
+		);
+	} );
+
+	test( 'storeSegments()', () => {
```

Comment:
I'm personally of the camp where this should be a `describe` block:
```js
describe( 'storeSegments()', () => {
```
And the `test` block should tell us what to expect from the test below:

```js
describe( 'storeSegments()', () => {
  test( 'should return the same output as setSegments', () => {
  ...
```
A pattern I've been playing with a little lately is to do all of the test action in a describe block and use a new test block to state each thing that we expect from the results (Using the next test as an example):

```js
describe( 'showVerticalsRequestError()', () => {
  const result = showSegmentsRequestError();

  test( 'should have a `type` attribute with value `NOTICE_CREATE`', () => {
  	expect( result.type ).toEqual( NOTICE_CREATE );	
  } );

  test( 'should have a notice attribute of plain object type', () => {
  	expect( isPlainObject( result.notice ) ).toEqual( true );
  } );

  test( 'should have a status attribute, on the notice object, with value `is-error`', () => {
  	expect( result.notice.status ).toEqual( 'is-error' );
  } );
} );
```
In this example the test set up is very slim, so it's hard to see the benefit of doing it this way, but you might imagine something like this (slightly contrived) example:

```js
describe( 'showVerticalsRequestError()', () => {
  const importerId = '12345';
  const importStatus = { id: importerId, type: 'someType' },
  const action = {
    importerId, 
    importStatus,
    type: IMPORTS_IMPORT_RECIEVE,
  };
  const existingImporterId = '54321';
  const existingImportStatus = { id: existingImporterId, type: 'someType' },
  const state = {
    [ existingImporterId ]: existingImportStatus,
  };
  const result = reducer( state, action );

  test( 'should add an importer item keyed by the given id, with a value of the given importerStatus', () => {
  	expect( result[ importerId ] ).toEqual( importStatus );	
  } );

  test( 'it does not affect existing importer items', () => {
  	expect( result[ existingImporterId ] ).toEqual( existingImporterStatus );	
  } );

```

Reply:
> I'm personally of the camp where this should be a describe block:

Reply:
In fact, I'm going to delete this test file as there are no real side effects. 

We're already testing our state methods, and the data layer is very well tested already.

Reply:
> But you speak my linguine.

I'm glad we see carb to carb :)